### PR TITLE
Added all the algorithms of trees data structure

### LINKED ISSUE #663: [Feature Request] Adding All the Basic Algorithms implemented in Trees
### Is your feature request related to a problem? Please describe.
A clear and concise description of what the different algorithms of trees are for easy one stop understanding

### Describe the solution you'd like
A clear and concise code structure for all the algo's required 

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by karanmaheshwari16**
### Description
I have added all the basic an most used algorithms used in the trees data structure in data structures and algorithms.
They are written in format simple to understand and have no bugs and work perfectly.

Fixes #663

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by karanmaheshwari16** [comment]
@ajay-dhangar please review this pull request

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
We only need the `.md` file for our documentation. Please delete the other files. like `.cpp`

**Comment by karanmaheshwari16** [comment]
> We only need the `.md` file for our documentation. Please delete the other files. like `.cpp`

sir these are the c++ files that represent the code.
Should i add a .md file in the current dir for the documentation?

otherwise all i have is the .cpp files for the code

**Comment by ajay-dhangar** [comment]
> > We only need the `.md` file for our documentation. Please delete the other files. like `.cpp`
> 
> sir these are the c++ files that represent the code. Should i add a .md file in the current dir for the documentation?
> 
> otherwise all i have is the .cpp files for the code

Only write code in `.md` file and delete all `.cpp` files

**Comment by karanmaheshwari16** [comment]
Done Sir 
Deleted all the .cpp files and integrated all the code in .md file

File: trees/tree.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
+# Tree Algorithms in C++
```
Comment (ajay-dhangar) [inline trees/tree.md]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update code based on suggestions

**Comment by karanmaheshwari16** [comment]
Done Sir

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
you write your file in wrong place like `trees/tree.md` but we need in `docs` folder so, can you write in docs folder

**Comment by karanmaheshwari16** [comment]
done👍

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**